File: oops_data_structure/Day9/AddressBook.java
class ContactPerson {

    private String firstName;
    private String lastName;
    private String address;
    private String city;
    private String state;
    private String zip;
    private String phoneNumber;
    private String email;

    public ContactPerson(String firstName, String lastName, String address,
                         String city, String state, String zip,
                         String phoneNumber, String email) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.address = address;
        this.city = city;
        this.state = state;
        this.zip = zip;
        this.phoneNumber = phoneNumber;
        this.email = email;
    }

    public void displayContact() {
        System.out.println(firstName + " " + lastName + ", " + address + ", "
                + city + ", " + state + " - " + zip
                + ", Phone: " + phoneNumber
                + ", Email: " + email);
    }
}
public class AddressBook {
    public static void main(String args[])

    {
        System.out.println("Welcome to Address Book Program");

        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter First Name:");
        String firstName = scanner.nextLine();

        System.out.println("Enter Last Name:");
        String lastName = scanner.nextLine();

        System.out.println("Enter Address:");
        String address = scanner.nextLine();

        System.out.println("Enter City:");
        String city = scanner.nextLine();

        System.out.println("Enter State:");
        String state = scanner.nextLine();

        System.out.println("Enter Zip:");
        String zip = scanner.nextLine();

        System.out.println("Enter Phone Number:");
        String phone = scanner.nextLine();

        System.out.println("Enter Email:");
        String email = scanner.nextLine();

        ContactPerson person = new ContactPerson(
                firstName, lastName, address, city, state, zip, phone, email
        );
        person.displayContact();
    }
}

File: web_intro/ObjectClass/AddressBook.js
// UC4: Find and Edit Contact in Address Book

class Contact {

    constructor(firstName, lastName, address,
                city, state, zip, phoneNumber, email) {

        this.firstName = firstName;
        this.lastName = lastName;
        this.address = address;
        this.city = city;
        this.state = state;
        this.zip = zip;
        this.phoneNumber = phoneNumber;
        this.email = email;
    }

    // Display Contact
    display() {

        return `
First Name  : ${this.firstName}
Last Name   : ${this.lastName}
Address     : ${this.address}
City        : ${this.city}
State       : ${this.state}
ZIP         : ${this.zip}
Phone Number: ${this.phoneNumber}
Email       : ${this.email}
`;
    }
}

// Address Book Array
let addressBook = [];

// Add Contacts
addressBook.push(
    new Contact(
        "Bharati",
        "Patra",
        "12 Gandhi Street",
        "Chennai",
        "TamilNadu",
        "600001",
        "9876543210",
        "abc.xyz@bridgelabz.co.in"
    )
);

addressBook.push(
    new Contact(
        "Rahul",
        "Sharma",
        "45 Nehru Road",
        "Mumbai",
        "Maharashtra",
        "400088",
        "9876541230",
        "rahul@bridgelabz.co"
    )
);

// Find Contact by First Name
let contact = addressBook.find(
    person => person.firstName === "Rahul"
);

// Edit Contact Details
if (contact) {

    contact.city = "Pune";
    contact.state = "Maharashtra";
    contact.phoneNumber = "9999999999";

    console.log("Contact Updated Successfully");

} else {

    console.log("Contact Not Found");
}

// Display Updated Address Book
addressBook.forEach(person => {
    console.log(person.display());
});